packages/client/src/runtime/core/model/UserArgs.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Input that flows from the user into the Client.
+ */
+export type UserArgs = {
+  [K in string]: UserArgsProp | UserArgsProp[]
+}
+
+type UserArgsProp = UserArgs | string | number | boolean | bigint | object | null | undefined
```

Comment:
The original type I could see previous code is `object | undefined`, here it expands the possibilities a lot, and I'm unsure if all are needed? (trying to think about it, but a quick chat would help probably)